src/transforms/compound.rs

```diff
@@ -37,8 +41,22 @@ impl TransformConfig for CompoundConfig {
     fn expand(
         &mut self,
     ) -> crate::Result<Option<(IndexMap<String, Box<dyn TransformConfig>>, ExpandType)>> {
-        if !self.nested.is_empty() {
-            Ok(Some((self.nested.clone(), ExpandType::Serial)))
+        let steps = &self.steps;
+        if !steps.is_empty() {
+            Ok(Some((
+                steps
+                    .iter()
+                    .enumerate()
+                    .map(|(i, step)| {
+                        let TransformStep { id, transform } = step;
+                        (
+                            id.as_ref().cloned().unwrap_or_else(|| i.to_string()),
+                            transform.to_owned(),
+                        )
+                    })
```

Comment:
I realize this is a corner case, but: what happens if a user specifies an integer ID that overlaps with one of the step index?  Does it just mess up metrics or would anything else break?

Reply:
Good catch! It actually overwrites the previous component with that ID.

Given the previous implementation also suffered that issue (duplicate IDs overwrite each other), I'll just note this as another thing to fix on https://github.com/timberio/vector/issues/8848 before this component is officially released.

This issue also affects the `route` transform.